common/remote/sockfile.cpp

```diff
@@ -2582,6 +2593,172 @@ void CRemoteFile::copyTo(IFile *dest, size32_t buffersize, ICopyFileProgress *pr
+                }
+                if (0 == maxRecs)
+                    break;
+                MemoryBuffer sendBuffer;
+                Owned<CRemoteFileIO> remoteIO = prepKeySend(sendBuffer);
+                sendBuffer.append((RemoteFileCommandType) RFCreadfilteredindex).append(first).append(remoteIO->getHandle()).append(filename).append(maxRecs);
+                rowDataBuffer.clear();
+                if (first)
+                {
+                    first = false;
+                    sendBuffer.append(keySize);
+                    segs.serialize(sendBuffer);
+                }
+                remoteIO->sendRemoteCommand(sendBuffer, rowDataBuffer);
+                unsigned recsGot;
+                rowDataBuffer.read(recsGot);
+                if (0 == recsGot)
+                {
+                    maxRecsPerRequest = 0;
+                    break; // end
+                }
+                totalGot += recsGot;
+                rowDataBuffer.read(sizeRemaining);
+            }
+        }
+        return false;
+    }
+    virtual unsigned __int64 getCount() override
+    {
+        return _checkCount((unsigned __int64)-1);
+    }
+    virtual unsigned __int64 getCurrentRangeCount(unsigned groupSegCount) override { UNIMPLEMENTED; }
+    virtual bool nextRange(unsigned groupSegCount) override { UNIMPLEMENTED; }
+    virtual void setKey(IKeyIndexBase * _key) override { UNIMPLEMENTED; }
+    virtual void setChooseNLimit(unsigned __int64 _chooseNLimit) override
+    {
+        chooseNLimit = _chooseNLimit;
+    }
+    virtual unsigned __int64 checkCount(unsigned __int64 limit) override
+    {
+        return _checkCount(limit);
+    }
+    virtual void serializeCursorPos(MemoryBuffer &mb) override { UNIMPLEMENTED; }
+    virtual void deserializeCursorPos(MemoryBuffer &mb) override { UNIMPLEMENTED; }
+    virtual unsigned querySeeks() const override { return 0; }
+    virtual unsigned queryScans() const override { return 0; }
+    virtual unsigned querySkips() const override { return 0; }
+    virtual unsigned queryNullSkips() const override { return 0; }
+    virtual const byte *loadBlob(unsigned __int64 blobid, size32_t &blobsize) override { UNIMPLEMENTED; }
+    virtual void releaseBlobs() override { /* TBD */ }
+    virtual void resetCounts() override { UNIMPLEMENTED; }
+
+    virtual void setLayoutTranslator(IRecordLayoutTranslator * trans) override { UNIMPLEMENTED; }
+    virtual void setSegmentMonitors(SegMonitorList &segmentMonitors) override { UNIMPLEMENTED; }
+    virtual void deserializeSegmentMonitors(MemoryBuffer &mb) override { UNIMPLEMENTED; }
+    virtual void finishSegmentMonitors() override { }
+    virtual bool lookupSkip(const void *seek, size32_t seekGEOffset, size32_t seeklen) override { UNIMPLEMENTED; }
+    virtual void append(IKeySegmentMonitor *segment) override
+    {
+        segs.append(segment);
+    }
+    virtual unsigned ordinality() const override { UNIMPLEMENTED; }
```

Comment:
yes could be, although not foreseen to be used yet. A couple of other seg methods could be implemented too.